src/object/tcb.c

```diff
@@ -1473,8 +1507,10 @@ exception_t decodeSetSpace(cap_t cap, word_t length, cte_t *slot, word_t *buffer
     }
 
 #ifdef CONFIG_KERNEL_MCS
-    cRootData = getSyscallArg(0, buffer);
-    vRootData = getSyscallArg(1, buffer);
+    word_t fhData    = getSyscallArg(0, buffer);
```

Comment:
I have to admit that I had to stare at this for a moment before remembering a reason, but I think it was due to wanting to maintain the same order of parameters as for non-MCS code.

Since the intention is to eventually have both the MCS and non-MCS code support fault handlers in the same way, in some sense I think changing the order of the parameters is unavoidable.

The options appear to me to be
1) change the order for MCS code now
2) change the order for MCS code in the future, when fault handlers are made consistent
3) change the order for non-MCS code in the future, when fault handlers are made consistent

I don't personally have a strong preference for any of these options.

Reply:
I'd go for option 3, considering you're changing non-MCS match MCS and not the other way round.